### PR TITLE
Add Plain Keyboard from KeyboardFlags.None

### PR DESCRIPTION
**Pull request by scastria**
### Description of Change ###

When creating an Entry for a username, for example, there should be NO possible Keyboard interference with the typing performed by the user.  No autocaps, spellcheck, suggestions, etc.  usernames are typically not english words so we don't want any chance of the Keyboard itself messing up what the user types.  Therefore, there should be a keyboard that has none of these features.  Currently, there is no obvious way of doing this except for creating a custom keyboard with KeyboardFlags.  However, KeyboardFlags has no option to choose None of the features.  There is NO 0 value in the KeyboardFlags enum.  Therefore, the developer must hardcode a 0 and then cast it to KeyboardFlags like this:

Keyboard.Create((KeyboardFlags)0)

This is not very appealing.  This pull request allows a developer to do this better in two ways:

Keyboard.Plain
Keyboard.Create(KeyboardFlags.None)

Also, the MSDN docs for a [Flags] based enum lists a best practice to include a None value assigned to 0.

### API Changes ###

Added KeyboardFlags.None enum value
Added Keyboard.Plain which uses KeyboardFlags.None for its implementation

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [ ] Consolidate commits as makes sense

### CODE REVIEW

**Comment by DavidKarlas** [comment]
You should probably add `Keyboard.Plain` to `keyboardTypes` in `KeyboardCoreGallery` constructor.

**Comment by DavidKarlas** [comment]
I would also like to note that this is very important to have for .xaml code... As @scastria said for entries with username or city name(non-english)...

**Comment by scastria** [comment]
You are right. I didn't see the keyboardTypes at the top of the gallery file. I will add that.

**Comment by scastria** [comment]
what does cla-required mean? Do I need to do anything further on this to help it get accepted and merged in? I am new to pull requests.

**Comment by rmarinho** [comment]
But this will need docs update, can you rebase and try run the docs bat on windows, or do 'make doc' on osx. have you checked this feature works in all platforms?

**Comment by scastria** [comment]
Sorry, I am still very new to GIT so I couldn't make the rebase work for me.  Therefore, I just updated the docs in my current branch as is.

**Comment by scastria** [comment]
I am not able to check that this feature works on Windows platforms.  However, I did visually inspect the code for the Windows platforms, and all platforms handle CustomKeyboard's the same way.  Check the KeyboardFlags for which bits are turned on and enable the platform specific equivalent feature.  Since the Plain keyboard has NO flag bits set, none of the platform specific equivalent features will be turned on.

**Comment by rmarinho** [comment]
👍

File: Xamarin.Forms.Core/Keyboard.cs [hunk 1]
```diff
@@ -21,8 +23,12 @@ internal Keyboard()
 		{
 		}
 
-		public static Keyboard Chat
+		public static Keyboard Plain
```
Comment (jassmith) [inline Xamarin.Forms.Core/Keyboard.cs]: Im not sure how I feel about Plain yet

**Comment by adrianknight89** [comment]
@scastria @jassmith I really like `KeyboardFlags.None`, but I have mixed feelings about `Keyboard.Plain`.

`Plain` is not a keyboard type. I'd rather use `Default` with `KeyboardFlags.None`.

**Comment by adrianknight89** [comment]
@scastria @jassmith Also filed a bug with regards to suggestions not being disabled on Android:

https://bugzilla.xamarin.com/show_bug.cgi?id=43811

**Pull request merged**